garden-backend-service/src/models/mdf_dataset.py

```diff
@@ -0,0 +1,20 @@
+from sqlalchemy import ForeignKey, String
+from sqlalchemy.dialects.postgresql import ARRAY
+from sqlalchemy.orm import Mapped, mapped_column
+from src.models.base import Base
+
+
+class Dataset(Base):
+    __tablename__ = "mdf-datasets"
+    id: Mapped[str] = mapped_column(primary_key=True)  # mdf SI subject id
+    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"))
+
+    flow_action_id: Mapped[str]
+
+    doi: Mapped[str]
+    source_id: Mapped[str]
+    version: Mapped[str]
+
+    previous_versions: Mapped[list[str]] = mapped_column(
```

Comment:
we should sync up on this and decide if we want to normalize this and make versions into its own table with a 1-many relationship or leave as an array -- a rule of thumb for using an array type vs a related table is whether or not we can always treat the array value in the field as a single entity vs if we may sometimes be interested in the subvalues of the array